controller/linodemachine_controller_helpers.go

```diff
@@ -684,16 +685,17 @@ func getDefaultInstanceConfig(ctx context.Context, machineScope *scope.MachineSc
 
 // createInstance provisions linode instance after checking if the request will be within the rate-limits
 // Note: it takes a lock before checking for the rate limits and releases it after making request to linode API or when returning from function
-func createInstance(ctx context.Context, logger logr.Logger, machineScope *scope.MachineScope, createOpts *linodego.InstanceCreateOptions) (*linodego.Instance, error) {
+func createInstance(ctx context.Context, logger logr.Logger, machineScope *scope.MachineScope, createOpts *linodego.InstanceCreateOptions) (*linodego.Instance, time.Duration, error) {
 	ctr := util.GetPostReqCounter(machineScope.TokenHash)
 	ctr.Mu.Lock()
 	defer ctr.Mu.Unlock()
 
 	if ctr.IsPOSTLimitReached() {
 		logger.Info(fmt.Sprintf("Cannot make more POST requests as rate-limit is reached (%d per %v seconds). Waiting and retrying after %v seconds", reconciler.SecondaryPOSTRequestLimit, reconciler.SecondaryLinodeTooManyPOSTRequestsErrorRetryDelay, reconciler.SecondaryLinodeTooManyPOSTRequestsErrorRetryDelay))
-		return nil, util.ErrRateLimit
+		return nil, ctr.RetryAfter(), util.ErrRateLimit
 	}
 
 	machineScope.LinodeClient.OnAfterResponse(ctr.ApiResponseRatelimitCounter)
-	return machineScope.LinodeClient.CreateInstance(ctx, *createOpts)
+	inst, err := machineScope.LinodeClient.CreateInstance(ctx, *createOpts)
+	return inst, time.Duration(reconciler.SecondaryLinodeTooManyPOSTRequestsErrorRetryDelay.Seconds()), err
```

Comment:
Is this time only useful if the error is non-nil? If so we should add a comment explaining that

Reply:
Sounds good, thanks for the clarification!